PdfJsViewerPlugin.inc.php

```diff
@@ -67,17 +68,27 @@ function articleCallback($hookName, $args) {
 		$templateMgr = TemplateManager::getManager($request);
 		if ($galley && $galley->getFileType() == 'application/pdf') {
 			$application = Application::getApplication();
+
+      $journal = $request->getJournal();            
+   
+      $pdfUrl = $journal->getPath() . '/article/download/' . $article->getBestArticleId() . '/' . $galley->getBestGalleyId();
```

Comment:
Manually concatenated URLs won't work when `disable_path_info` is turned `On` in `config.inc.php`.

Reply:
Is there an URL constructing function in pkp-lib I should use? 
I see the galley templates are using the `url` function, but that one does not allow the base URL to be changed.

Reply:
@sean-xiao-zhao7, all URLs are currently generated through `PKPRouter::url`, a.k.a. `Request::url`, a.k.a. `{url}` in Smarty. I'm curious why this case needs a specific change when the system's behavior regarding `index.php` should be standardized throughout?

Reply:
@mjordan, maybe this is something you're familiar with?

Reply:
My ezproxy are quite rusty, but are we sure there isn't an exproxy config directive that will prevent the cross origin error? @calvinm might know, but he'll need some additional info I suspect.

Reply:
Sorry, didn't read that very closely. It's not an exproxy issue, it's on the OJS side.

Reply:
I'm still not really clear what's happening, to be honest. @sean-xiao-zhao7, can you describe at a high level what you're expecting, and what's happening instead?

OJS's built-in url generation function will either include the `index.php` or not, depending on the configuration; but it should do so consistently, so a proposed change in one place here has me confused.

Reply:
Yep. On article view page, the iframe has `src` of the form `https://ezproxy_ojs_url/plugins/generic/pdfJsViewer/pdf.js/web/viewer.html?file=non_ezproxy_ojs_url/index.php/journal_name/article/download/12345/12345`
The HTTP parameter for `file` is the `$pdfUrl` variable generated by `url` witihn the `articleGalley.tpl` template.
Not sure why it does not detect the new base URL.

Reply:
So essentially ezproxy is rewriting the first part of the URL but not the second (urlencoded parameter), correct?

Reply:
yes exactly

Reply:
I don't know much at all about ezproxy, but as I understand it, there are different modes instructing it to rewrite different things (e.g. a configuration option that'll tell it to rewrite Javascript etc). OJS itself is working properly and consistently as far as I can see, and your proposal here would break that. Is there maybe an ezproxy configuration setting that instructs it to look for urlencoded URLs for rewriting?

Reply:
That's a good question, I'm not expert on EZProxy either, but I can ask around

Reply:
My experience with ezproxy is more on the library side, not the publisher side.  In cases where it wouldn't work with Javascript, I would specify HJ, or DJ on the host or domain respectively in the ezproxy stanza.

If that doesn't work you may need to manually add a search/replace in the ezproxy stanza to look for the file=non_ezproxy_ojs_url and replace it with file=ezproxy_ojs_url.  This would look like this in the ezproxy stanza:

Find file=non_ezproxy_ojs_url 
Replace file=ezproxy_ojs_url

This is how I would make it work on the ezproxy side, but again, I'm not sure how to make it work on the OJS side.

Reply:
thanks all, we were able to adjust our EZProxy instance to make this work.
There was a typo in one of the stanzas; when we fixed it, that stanza and all other OJS related stanzas started working, not sure why one stanza affected the others.